geode-core/src/main/java/org/apache/geode/management/internal/configuration/mutators/ConfigurationMutator.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.geode.management.internal.configuration.mutators;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.cache.configuration.CacheConfig;
+import org.apache.geode.cache.configuration.CacheElement;
+
+/**
+ * Defines the behavior to mutate a configuration change into a pre-existing cache config from a
+ * locator
+ * {@link org.apache.geode.distributed.ConfigurationPersistenceService}. Created with an object of
+ * type {@link CacheElement}, which represents the configuration change.
+ */
+@Experimental
+public interface ConfigurationMutator<T extends CacheElement> {
+  void add(T config, CacheConfig existing);
```

Comment:
Why would any mutation operation not return the "new" `CacheConfig`? I think it would be good practice NOT to modify-in-place

Reply:
It would be a good practice not to modify in-place, but don't think it's a bad practice to modify in-place either. If you think about how this could be used, it might even be cleaner to modify in place, so you don't have to declare more variables.

Reply:
So we are not concerned about concurrency? *COULD* there be any potential that multiple threads could change the same  CacheConfig and instance and modify-in-place? How would the system react if two threads were to amend the same CacheConfig object and ask it to persist it... If concurrency is not a concern, this question is mute

Reply:
The mutators are used technically outside the scope of the methods that actually operate on the common persisted config on the locator. We "atomically" get a CacheConfig object representation of the persisted config (essentially a copy), mutate it, and then it gets written back -- if you look at the way this is used in LocatorClusterManagementService it would make this clear, since the underlying method that does this acquires a lock. So in this case it's thread safe whether you modify in-place or not.